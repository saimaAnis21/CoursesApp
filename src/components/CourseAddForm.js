import React, { useState, useEffect } from 'react';
import fetchCategoryNames from '../logic/fetchCategoryNames';
// import { GetAuthToken } from '../logic/localStorage';
import CategoryNamesSelect from './CategoryNamesSelect';
import '../styles.css';

const CourseAddForm = () => {
  const [title, setTitle] = useState();
  const [duration, setDuration] = useState();
  const [category, setCategory] = useState();
  const [categorynames, setCategoryNames] = useState([{ id: 0, name: '--' }]);

  const getData = async () => {
    const response = await fetchCategoryNames();
    setCategoryNames(response);
    // console.log(categorynames);
  };

  useEffect(() => {
    // console.log(GetAuthToken());
    getData();
  });

  const inputChange = (e) => {
    if (e.target.name === 'title') {
      setTitle(e.target.value);
    } else if (e.target.name === 'duration') {
      setDuration(e.target.value);
    } else if (e.target.name === 'category') {
      setCategory(e.target.value);
    }
  };

  const btnClick = (e) => {
    e.preventDefault();
    console.log(title + duration + category);
  };

  return (
    <div className=" p-20 txt-align mx-auto mt-50 w-50 border border-primary rounded">
      <form className="">
        <div className="mt-10 mb-10 d-flex flex-column align-items-center mx-auto w-75 mx-auto">

          <div className="d-flex justify-content-between w-75 mb-10">
            <span>Title:</span>
            <input
              type="text"
              id="title"
              name="title"
              onChange={(e) => inputChange(e)}
            />
          </div>

          <div className="d-flex justify-content-between w-75 mb-10">
            <span>Duration:</span>
            <input
              type="text"
              id="duration"
              name="duration"
              onChange={(e) => inputChange(e)}
            />
          </div>

          <div className="d-flex justify-content-between w-75 mb-10">
            <span>Category:</span>
            <CategoryNamesSelect />            
          </div>

          <button
            type="submit"
            className="btn btn-primary btn-lg btn-block"
            onClick={(e) => btnClick(e)}
          >
            Create Course
          </button>
        </div>

      </form>
    </div>
  );
};

export default CourseAddForm;
