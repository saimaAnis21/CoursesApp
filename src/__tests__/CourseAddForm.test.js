import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import CourseAddForm from '../components/CourseAddForm';

const store = createStore(() => [], {});
const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

Wrapper.propTypes = {
  children: PropTypes.shape({}).isRequired,
};

describe('Course Add Form', () => {
  it('renders correctly', () => {
    const tree = render(<CourseAddForm />, { wrapper: Wrapper });
    expect(tree).toMatchSnapshot();
  });
});
