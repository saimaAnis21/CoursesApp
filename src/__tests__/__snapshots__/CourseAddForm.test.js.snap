// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Add Form renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" p-20 txt-align mx-auto mt-10 "
      >
        <h1>
          Create a course!
        </h1>
        <p
          class="text-danger"
        />
        <form
          class=""
        >
          <div
            class="mt-10 mb-10 d-flex flex-column "
          >
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="basic-addon1"
                >
                  Course Title
                </span>
              </div>
              <input
                aria-describedby="basic-addon1"
                class="form-control"
                id="title"
                name="title"
                placeholder="e.g. Crash Course in Java"
                type="text"
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="basic-addon1"
                >
                  Duration(hrs)
                </span>
              </div>
              <input
                aria-describedby="basic-addon1"
                class="form-control"
                id="duration"
                name="duration"
                placeholder="e.g. 1.0 or 2.5"
                type="text"
              />
            </div>
            <div
              class="d-flex justify-content-between w-75 mb-10"
            >
              <div
                class="input-group mb-3 "
              >
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text "
                  >
                    Categories
                  </span>
                </div>
                <select
                  class="custom-select w-50"
                  id="category"
                  name="category"
                >
                  <option
                    value="0"
                  >
                    --
                  </option>
                  ;
                </select>
              </div>
            </div>
            <button
              class="btn btn-dark btn-sm btn-block"
              type="submit"
            >
              Create Course
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" p-20 txt-align mx-auto mt-10 "
    >
      <h1>
        Create a course!
      </h1>
      <p
        class="text-danger"
      />
      <form
        class=""
      >
        <div
          class="mt-10 mb-10 d-flex flex-column "
        >
          <div
            class="input-group mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
                id="basic-addon1"
              >
                Course Title
              </span>
            </div>
            <input
              aria-describedby="basic-addon1"
              class="form-control"
              id="title"
              name="title"
              placeholder="e.g. Crash Course in Java"
              type="text"
            />
          </div>
          <div
            class="input-group mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
                id="basic-addon1"
              >
                Duration(hrs)
              </span>
            </div>
            <input
              aria-describedby="basic-addon1"
              class="form-control"
              id="duration"
              name="duration"
              placeholder="e.g. 1.0 or 2.5"
              type="text"
            />
          </div>
          <div
            class="d-flex justify-content-between w-75 mb-10"
          >
            <div
              class="input-group mb-3 "
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text "
                >
                  Categories
                </span>
              </div>
              <select
                class="custom-select w-50"
                id="category"
                name="category"
              >
                <option
                  value="0"
                >
                  --
                </option>
                ;
              </select>
            </div>
          </div>
          <button
            class="btn btn-dark btn-sm btn-block"
            type="submit"
          >
            Create Course
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
